Communicate with certutil via PEM instead of DER.

JeremyRand opened the pull request:
This should reduce the risk of concatenated certs, and other encoding issues.

Nantaya555 commented:
Happy day

hlandau commented:
ACK https://github.com/namecoin/tlsrestrictnss/pull/6/commits/4682d9cc4222cfa8d7f91f94aeffc59f8e6202ba

JeremyRand committed:
Communicate with certutil via PEM instead of DER.

This should reduce the risk of concatenated certs, and other encoding issues.

JeremyRand committed:
Fix detection of certType == certTypeCrossSigned for DER certs.

A cross-signed CA will have a Subject CommonName that's identical to the original cert's Subject CommonName.  It won't have a prefix like the nickname will.

hlandau commented:
ACK https://github.com/namecoin/tlsrestrictnss/pull/6/commits/04a5f41729ef0b387546e048c0e7a200498da65d
